onnx/test/shape_inference_test.py

```diff
@@ -2980,6 +2980,14 @@ def test_einsum_transpose(self):  # type: () -> None
             [],)
         self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, (None, None))])  # type: ignore
 
+    def test_einsum_outer_prod(self):  # type: () -> None
+        graph = self._make_graph(
+            [('x', TensorProto.FLOAT, (3, 5)),
+             ('y', TensorProto.FLOAT, (7, 9))],
+            [make_node('Einsum', ['x', 'y'], ['z'], equation='ij,ab->ijab')],
+            [],)
+        self._assert_inferred(graph, [make_tensor_value_info('z', TensorProto.FLOAT, (None, None, None, None))])  # type: ignore
```

Comment:
So, the inference doesn't currently infer the output-shape of (3,5,7,9), even though it has the necessary information to infer it? We don't have to do this in the current PR/release, but I think we should make the shape-inference more complete to infer the dimensions where possible.

Reply:
So actually I did not look into shape inference since Einsum could results in many different operations. I'm not sure if it's okay to have shape inference supported partially, based on the equation. If you think otherwise, we can look into supporting some cases.

Reply:
As I mentioned above, let us not do it for this PR or this release. I can help do this after the release.

Reply:
Thanks @gramalingam